src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/legacy/service/LegacyPeriodLengthService.kt

```diff
@@ -9,20 +9,60 @@ import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.jpa.entity.audit
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.jpa.enum.EntityChangeStatus
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.jpa.enum.EntityStatus
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.jpa.repository.PeriodLengthRepository
+import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.jpa.repository.SentenceRepository
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.jpa.repository.audit.PeriodLengthHistoryRepository
+import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.legacy.controller.dto.LegacyCreatePeriodLength
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.legacy.controller.dto.LegacyPeriodLength
+import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.legacy.controller.dto.LegacyPeriodLengthCreatedResponse
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.legacy.controller.dto.NomisPeriodLengthId
 import uk.gov.justice.digital.hmpps.hmppsremandandsentencingapi.service.ServiceUserService
 import java.util.UUID
 
 @Service
-class LegacyPeriodLengthService(private val periodLengthRepository: PeriodLengthRepository, private val periodLengthHistoryRepository: PeriodLengthHistoryRepository, private val serviceUserService: ServiceUserService) {
+class LegacyPeriodLengthService(
+  private val periodLengthRepository: PeriodLengthRepository,
+  private val periodLengthHistoryRepository: PeriodLengthHistoryRepository,
+  private val serviceUserService: ServiceUserService,
+  private val sentenceRepository: SentenceRepository,
+) {
+  fun create(periodLength: LegacyCreatePeriodLength): LegacyPeriodLengthCreatedResponse {
+    val sentenceEntities = sentenceRepository.findBySentenceUuid(periodLength.sentenceUuid)
+    val firstSentenceEntity = sentenceEntities.firstOrNull()
+      ?: throw EntityNotFoundException("No sentence found with UUID ${periodLength.sentenceUuid}")
+    val isManyCharges = sentenceEntities.first().charge.appearanceCharges.size > 1
```

Comment:
already retrieved and checked its null
```suggestion
    val isManyCharges = firstSentenceEntity.charge.appearanceCharges.size > 1
```